packages/core-components/api-report.md

```diff
@@ -699,6 +700,23 @@ export function MissingAnnotationEmptyState(props: Props_3): JSX.Element;
 // @public (undocumented)
 export type MissingAnnotationEmptyStateClassKey = 'code';
 
+// @public
+export const MobileSidebar: (props: MobileSidebarProps) => JSX.Element | null;
+
+// @public
+export const MobileSidebarContext: React_2.Context<MobileSidebarContextType>;
```

Comment:
Can we keep this and the type private? If it really needs to be public API we can export a provider component and a hook instead

Reply:
I tried a few things, but I couldn't really find a way to do this. The reason why I'm using the context is just because I want to share the state with the underlying `SidebarGroup`s (which are filtered in the `MobileSidebar`). Normally you would just pass down the state & setter as a `prop`. This doesn't work here though. To achieve this we would have to modify the props of the children, which are actually `SidebarGroup`s - which obviously requires a cloning of the elements, as the props should be read-only. I tried the following, but it had no affect (used code): 
```
  let sidebarGroups = useElementFilter(children, elements =>
    elements
      .getElements()
      .filter(child => child.type === SidebarGroup)
      .map(child =>
        React.cloneElement(child, {
          selectedMenuItemIndex,
          setSelectedMenuItemIndex,
        }),
      ),
  );
```

My issue with creating a custom hook: I'm using the context, because I want exactly the things the `useContext` hook provides me - the hook would basically be `useContext(...)`, which seems a bit odd. Additionally whole `MobileSidebar` & `SidebarGroup` component are already quite complicated & abstract - building another abstraction just to avoid an export feels kind of weird & more like another workaround.